src/models/sales.ts

```diff
@@ -0,0 +1,97 @@
+import { Model, DataTypes, CreationOptional } from 'sequelize';
+import { connection } from '../database';
+import Product from './products';
+
+class Sale extends Model {
+  declare id: number;
+  declare fullName: string;
+  declare deliveryDate: string;
+  declare direction: string;
+  declare contactNumber: number;
+  declare deliveryTime: string;
+  declare deliveryCost: number;
+  declare tax: number;
+  declare totalWithTax: number;
+  declare amount: number;
+  declare surprise: boolean;
+  declare forwarding: boolean;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+}
+
+Sale.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      allowNull: false,
+      primaryKey: true,
+    },
+    fullName: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    deliveryDate: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    direction: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    contactNumber: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+    },
+    deliveryTime: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    deliveryCost: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+    },
+    tax: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+    },
+    totalWithTax: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+    },
+    productId: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+    },
+    amount: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+    },
+    surprise: {
+      type: DataTypes.BOOLEAN,
+      allowNull: false,
+    },
+    forwarding: {
+      type: DataTypes.BOOLEAN,
+      allowNull: false,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+    },
+  },
+  {
+    sequelize: connection,
+    modelName: 'Sale',
+    tableName: 'sales',
+  },
+);
+
+Product.hasMany(Sale);
+Sale.belongsTo(Product);
+
```

Comment:
Same associations issue here. We put each side of the association declaration in te corresponding module. Product.hasMany(Sale) should be in the Product model and not here. That would give you a problem if you just put that there, but you can solve that issue by declaring associations inside an `associate` method inside that model, and then calling that method only in the index file of the models folder once all models have been loaded